### ISSUE TITLE
lita, LANG, and US-ASCII

### ISSUE THREAD

**Issue by directionless** [litaio/chef-lita#28]
Hi Folks! I ran into an issue with encoding. I'm not really sure what the *correct* solution is, though I found a workaround. I'm opening this, since maybe folks have a suggestion, and if not, future people may appreciate it. Pretty sure https://github.com/litaio/lita-slack/issues/42 is an instance of this. 

Anyhow, on Ubuntu 12.04, using chef-lita, and runit, with the slack adapters. If users have non-ascii characters in their name, it causes lita-slack to crash:

```
2016-02-01_19:01:25.64141 /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/core_ext/object/json.rb:34:in `encode': "\xC3" on US-ASCII (Encoding::InvalidByteSequenceError)
2016-02-01_19:01:25.64169 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/core_ext/object/json.rb:34:in `to_json'
2016-02-01_19:01:25.64189 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/core_ext/object/json.rb:34:in `to_json_with_active_support_encoder'
2016-02-01_19:01:25.64201 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/json/encoding.rb:57:in `to_json'
2016-02-01_19:01:25.64215 	from /opt/lita/vendor/ruby/2.1.0/gems/json-1.8.3/lib/json/common.rb:223:in `generate'
2016-02-01_19:01:25.64235 	from /opt/lita/vendor/ruby/2.1.0/gems/json-1.8.3/lib/json/common.rb:223:in `generate'
2016-02-01_19:01:25.64251 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/json/encoding.rb:101:in `stringify'
2016-02-01_19:01:25.64261 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/json/encoding.rb:35:in `encode'
2016-02-01_19:01:25.64276 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/json/encoding.rb:22:in `encode'
2016-02-01_19:01:25.64292 	from /opt/lita/vendor/ruby/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/core_ext/object/json.rb:37:in `to_json_with_active_support_encoder'
2016-02-01_19:01:25.64307 	from /opt/lita/vendor/ruby/2.1.0/gems/multi_json-1.11.2/lib/multi_json/adapters/json_common.rb:19:in `dump'
2016-02-01_19:01:25.64326 	from /opt/lita/vendor/ruby/2.1.0/gems/multi_json-1.11.2/lib/multi_json/adapter.rb:25:in `dump'
2016-02-01_19:01:25.64336 	from /opt/lita/vendor/ruby/2.1.0/gems/multi_json-1.11.2/lib/multi_json.rb:136:in `dump'
2016-02-01_19:01:25.64351 	from /opt/lita/vendor/ruby/2.1.0/gems/lita-slack-1.7.2/lib/lita/adapters/slack/rtm_connection.rb:89:in `payload_for'
2016-02-01_19:01:25.64366 	from /opt/lita/vendor/ruby/2.1.0/gems/lita-slack-1.7.2/lib/lita/adapters/slack/rtm_connection.rb:104:in `safe_payload_for'
2016-02-01_19:01:25.64382 	from /opt/lita/vendor/ruby/2.1.0/gems/lita-slack-1.7.2/lib/lita/adapters/slack/rtm_connection.rb:62:in `block (2 levels) in send_messages'
2016-02-01_19:01:25.64404 	from /opt/lita/vendor/ruby/2.1.0/gems/lita-slack-1.7.2/lib/lita/adapters/slack/event_loop.rb:10:in `block in defer'
2016-02-01_19:01:25.64421 	from /opt/lita/vendor/ruby/2.1.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:1067:in `call'
2016-02-01_19:01:25.64442 	from /opt/lita/vendor/ruby/2.1.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:1067:in `block in spawn_threadpool'
```

Though `/etc/default/locale` is set to en_US.UTF-8, so I'm not sure where we're getting downgraded to ascii.

I worked around this, by whomping the runit env in my wrapper cookbook:
```ruby
lita_runit_env = {
  'HOME' => node['lita']['install_dir'],
  'PATH' => [node['languages']['ruby']['bin_dir'],
    node['languages']['ruby']['gem_bin'],
    node['languages']['ruby']['ruby_dir']
  ].join(':'),
  'LANG' => 'en_US.UTF-8',
}
node.default['lita']['runit']['env'] = lita_runit_env
```

**Comment by jimmycuadra** [comment]
This would be a good addition, but we should make the value configurable with an attribute.

**Comment by directionless** [comment]
What do you think about pushing the whole env straight into node attributes? It seems like it would be easier to override parts as needed. I saw the comment, but wasn't sure what you were getting at with it.

**Comment by jimmycuadra** [comment]
Personally, I don't have a preference one way or the other, as long as the cookbook doesn't lock the operator into values they can't change.

**Comment by directionless** [comment]
I'm working on https://github.com/directionless/chef-lita/tree/updates But I have one more bug I want to work in

**Comment by harlanbarnes** [comment]
Thanks for taking a stab at this But there is some history here. You are actually about to revert a bug fix I made based on #26. There's a derived attributes issue here that we are trying to avoid. (Otherwise those using wrapper scripts and non-defaults have to change multiple attributes.)

What you did in your wrapper script is actually correct, in my opinion. If you want to add a LANG setting to the default, I don't see any problem with that.

**Comment by directionless** [comment]
Ah well, sounds like there's no clean answer then. I will *not* PR it. Thank you!
